client/my-sites/pages/blog-posts-page.jsx

```diff
@@ -37,21 +69,43 @@ const BlogPostsPage = React.createClass( {
 				</div>;
 		}
 
-		let frontPageIcon = null;
-		if ( this.props.frontPageType === 'posts' ) {
-			frontPageIcon = <Gridicon icon="house" size={ 18 } />;
-		}
-
 		return (
 			<CompactCard className="pages__blog-posts-page">
 				{ notUsedLabel }
 				<span className="pages__blog-posts-page__title" href="">
-					{ frontPageIcon }
+					{ isCurrentlySetAsHomepage ? <Gridicon icon="house" size={ 18 } /> : null }
 					{ this.translate( 'Blog Posts' ) }
 				</span>
 				<div className="pages__blog-posts-page__info">
-					{ this.translate( 'Your latest posts' ) }
+					{
+						isCurrentlySetAsHomepage
+						? this.translate( 'Your latest posts, shown on homepage' )
+						: this.translate( 'Your latest posts' )
+					}
 				</div>
+				{
+					shouldShowPageActions
+					? <div>
```

Comment:
Check out `<EllipsisMenu>` which does exactly this: https://wpcalypso.wordpress.com/devdocs/design/ellipsis-menu